third_party/terraform/tests/resource_cloudiot_device_registry_update_test.go

```diff
@@ -0,0 +1,162 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccCloudIoTRegistry_update(t *testing.T) {
+	t.Parallel()
+
+	registryName := fmt.Sprintf("psregistry-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudIoTRegistryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudIoTRegistryBasic(registryName),
+			},
+			{
+				ResourceName:      "google_cloudiot_registry.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccCloudIoTRegistryExtended(registryName),
+			},
+			{
+				ResourceName:      "google_cloudiot_registry.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccCloudIoTRegistryBasic(registryName),
+			},
+			{
+				ResourceName:      "google_cloudiot_registry.foobar",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckCloudIoTRegistryDestroy(s *terraform.State) error {
```

Comment:
You shouldn't need this since you can just use the one from the generated test

Reply:
Nice! I didn't know about this. Will have a closer look at the generated provider

Reply:
done